lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -343,15 +350,33 @@ bool SslClientConnection::connectSocket() {
 
   ERR_clear_error();
 
-  int ret = SSL_connect(_ssl);
+  int ret = -1;
+  int errorDetail = -1;
+
+  if (_isTelemetrics) {
+    auto start = steady_clock::now();
+    while ((ret = SSL_connect(_ssl)) == -1) {
+      errorDetail = SSL_get_error(_ssl, ret);
+      if (_isInterrupted) {
+        break;
+      }
+      auto end = steady_clock::now();
+      if ((errorDetail != SSL_ERROR_WANT_READ &&
+           errorDetail != SSL_ERROR_WANT_WRITE) ||
+          duration_cast<seconds>(end - start).count() >= _connectTimeout) {
+        break;
+      }
+    }
+  } else {
+    ret = SSL_connect(_ssl);
+    errorDetail = SSL_get_error(_ssl, ret);
+  }
 
   if (ret != 1) {
     _errorDetails.append("SSL: during SSL_connect: ");
 
-    int errorDetail;
     long certError;
 
-    errorDetail = SSL_get_error(_ssl, ret);
     if ((errorDetail == SSL_ERROR_WANT_READ) ||
         (errorDetail == SSL_ERROR_WANT_WRITE)) {
       return true;
```

Comment:
If we drop out here and `_isTelemetrics` is `true`, we don't clean up the socket flags.
If there are other premature exits from this function, then they may also miss the cleanup.